### PR TITLE
[Notifier] [BC BREAK] Change constructor signature for Mattermost and Esendex transport

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 5.x, but BC BREAK for experimental bridge
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | ---
| License       | MIT
| Doc PR        | ---

Based on https://github.com/symfony/symfony/pull/39428#issue-535936925

cc @odolbeau as you provided the bridge

### CODE REVIEW

File: src/Symfony/Component/Notifier/Bridge/Esendex/EsendexTransport.php [hunk 1]
```diff
@@ -68,18 +70,20 @@ protected function doSend(MessageInterface $message): SentMessage
         }
 
         $response = $this->client->request('POST', 'https://'.$this->getEndpoint().'/v1.0/messagedispatcher', [
-            'auth_basic' => $this->token,
+            'auth_basic' => sprintf('%s:%s', $this->email, $this->password),
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Esendex/EsendexTransport.php:70]: Before, this was assembled in the factory

File: src/Symfony/Component/Notifier/Bridge/Esendex/EsendexTransport.php [hunk 2]
```diff
@@ -44,7 +46,7 @@ public function __construct(string $token, string $accountReference, string $fro
 
     public function __toString(): string
     {
-        return sprintf('esendex://%s', $this->getEndpoint());
+        return sprintf('esendex://%s?accountreference=%s&from=%s', $this->getEndpoint(), $this->accountReference, $this->from);
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Esendex/EsendexTransport.php:46]: I also updated the `__toString()` method to display `accountreference` and `from`, like in the other bridges

**Review by odolbeau — APPROVED** [review]

**Comment by OskarStark** [comment]
Ready to merge from my side

File: src/Symfony/Component/Notifier/Bridge/Esendex/CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,14 @@
 CHANGELOG
 =========
 
+5.3.0
+-----
+
+* [BC BREAK] Changed signature of `EsendexTransport::__construct()` method from:
```
Comment (jderusse) [inline src/Symfony/Component/Notifier/Bridge/Esendex/CHANGELOG.md:8]: Shouldn't it be reported in `UPGRADE-5.3.md`

File: src/Symfony/Component/Notifier/Bridge/Mattermost/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,14 @@
 CHANGELOG
 =========
 
+5.3.0
+-----
+
+* [BC BREAK] Changed signature of `MattermostTransport::__construct()` method from:
```
Comment (jderusse) [inline src/Symfony/Component/Notifier/Bridge/Mattermost/CHANGELOG.md:8]: same

**Review by jderusse — COMMENTED** [review]
Please rebase to fix conflicts

File: src/Symfony/Component/Notifier/Bridge/Esendex/CHANGELOG.md (hunk 3 above)
Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/Esendex/CHANGELOG.md:8]: Not sure 🧐

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @OskarStark.

**Pull request merged**